src/components/SearchClustersPage.vue

```diff
@@ -181,31 +187,34 @@ export default {
     searchFiltersEnabled() {
       return Boolean(this.$route.query[QueryParameters.SearchFiltersEnabled])
     },
-    searchApiQueryParameters() {
-      const pageNumber = this.paginationCurrentPage - 1;
-      const orderBy = this.orderByValue;
-      const filters = this.searchFiltersEnabled
-        ? serializeFilters(this.searchFilters)
-        : undefined;
-      return {
-        text: this.searchText,
-        skip: this.paginationPerPage * pageNumber,
-        limit: this.paginationPerPage,
-        orderBy,
-        filters
-      }
-    }
   },
   watch: {
-    searchApiQueryParameters: {
+    '$route.query': {
```

Comment:
It is better to have the watcher listen on a `query` property. This creates a better separation of concern: the function that send a request to a service/API does not need to know how the query is calculated and how and when it is updated. All it needs to know that it is called whenever this query changes. Separation of concern makes it easier to work with code and test it.

This code has been changed and it already has two bugs:
 * when the value passed to the  `paginationPerPage` component property is changed, this watcher is not called. And since `paginationList.perPage` variable never tracks the  property (explained above), this watcher is not called either when the property is changed.
* if the page query parameters include parameters that are not needed for this particular query and these parameters change, this watcher will be called with the same query it was called with before.

Computed properties are a great feature of `vue.js`, we should use them! :)

Reply:
I see your point, but it was quite difficult to follow on the code.. In terms of readability, using just one watcher function has the advantage that we get in one place all the code to deliver the list of clusters :)
Also, I don't see a scenario where we change the property paginationPerPage outside of this component, I suggest to remove is from the props list. Your second point is right, indeed.

Reply:
I appreciate that this might look different from what we are used to in impresso, but once you are used to it, understanding the code actually becomes much easier due to separation of concern (think about it as worrying about one problem at a time rather than multiple problems in parallel). Also having one problem per function / component lifecycle makes it less prone to bug attack.

I don't mind making `perPage` a constant rather than a property, but I do recommend having a computed property for the `service query` and a watcher that watches for changes in this `service query` and is triggered only when something there is changed. It is a good practice for other more complex pages.

Reply:
`searchApiQueryParameters` added back, with a simple watcher and an additional method `loadClusters()`